### ISSUE TITLE
CJK font rendering mixing up fonts

### ISSUE THREAD

**Issue by ndupraz** [notepad-plus-plus/notepad-plus-plus#8171]
There seems to be a font rendering issue for CJK characters, using the default settings, they are rendered using a different set of fonts, instead of a single one. Not sure how the font selection operates under Windows, but the issue isn't reproducible under Microsoft's notepad. Is there supposed to be a way to select or force a specific font for rendering?

Notepad++ v7.8.5 (64-bit)
OS Name : Windows 10 Enterprise LTSC 2019 (64-bit) 
OS Version : 1809
OS Build : 17763.1192

Attached is my system font list (relevant ones are Google Noto CJK, Adobe Source Han and GNU Unifont along with the defaults Microsoft CJK) and a screenshot at several font sizes.

![cjk](https://user-images.githubusercontent.com/64301472/80273394-3c545480-8704-11ea-96ef-5da23e817e2e.png)
[fonts.txt](https://github.com/notepad-plus-plus/notepad-plus-plus/files/4532507/fonts.txt)

**Comment by NightFurySL2001** [comment]
Yes there should be a setting for that, try searching Settings-Style Configurator, you should see this ![Style Configurator window](https://user-images.githubusercontent.com/33471049/80280203-40e53100-8735-11ea-895e-4af9a3ce633a.png). Under Default (stylers.xml) > Global Styles > Global override, on the right panel, Font Style > Font name, choose a Chinese font. (I'm using [Huninn](https://github.com/justfont/open-huninn-font) for checking issues so please choose another, like Source Han Sans/Noto CJK). Another suggestion is to use [Sarasa Term from Sarasa Gothic](https://github.com/be5invis/Sarasa-Gothic), which is the default font I use daily for coding usage with ligatures (if you know how to open those in Notepad++).

The issue in your pic is due to the font choice only support those characters (maybe a Japanese font), and then slowlu falling back through the list. (门 is in Japanese variant, 艺/历 is in Traditional Chinese variant). The key part isn't what fonts are installed, it's what the font substitution/font link (HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\FontLink\SystemLink) in your system works.

**Comment by ndupraz** [comment]
Specifying an universal font effectively solves the issue, but I'm still unsure about which font order happens to be selected by default or of how to change it. Is there a Windows equivalent for freedesktop's [fontconfig configuration](https://www.freedesktop.org/software/fontconfig/fontconfig-user.html)? Here is a copy of my system's font substitution register entries.

![hanzi](https://user-images.githubusercontent.com/64301472/80284108-f83a7180-874e-11ea-8e64-e0aa2962396e.png)
[fontlink.txt](https://github.com/notepad-plus-plus/notepad-plus-plus/files/4533405/fontlink.txt)

**Comment by NightFurySL2001** [comment]
Which font did your editor used? I would guess that the font stack it fallback-ed with is the Microsoft Sans Serif stack, but it acted differently on my system too. You will have to ask the creator @sasumner  on here.
